utils/install.py

```diff
@@ -0,0 +1,1636 @@
+                        )
+                else:
+                    if ipath not in fname:
+                        fname = join(ipath, fname)
+                # replace GNUSparseFile.0 with nothing
+                fname = fname.replace("/GNUSparseFile.0", "")
+                # Don't append system directories
+                if (not is_default_path(args)) and isdir(fname):
+                    continue
+                env_file.write("#" + fname + "\n")
+                logging.debug("Appending:%s", fname)
+        else:
+            logging.warning("Unable to write to env file")
+
+    if remove_finished:
+        remove(from_path)
+
+
+# https://stackoverflow.com/questions/1868714/
+# how-do-i-copy-an-entire-directory-of-files-
+# into-an-existing-directory-using-pyth
+def copytree(src, dst, symlinks=True, ignore=None):
+    if not exists(dst):
+        makedirs(dst)
+        shutil.copystat(src, dst)
+    lst = listdir(src)
+    if ignore:
+        excl = ignore(src, lst)
+        lst = [x for x in lst if x not in excl]
+    for item in lst:
+        s = join(src, item)
+        d = join(dst, item)
+        if symlinks and islink(s):
+            if lexists(d):
+                remove(d)
+            symlink(readlink(s), d)
+        elif isdir(s):
+            copytree(s, d, symlinks, ignore)
+        else:
+            shutil.copy2(s, d)
+
+
+class VersionString:
+    def __init__(self, version):
+        self.version = version
+
+    def __str__(self):
+        return self.version
+
+    def __repr__(self):
+        return "VersionString:" + self.version
+
+    def _preprocess(self, v, separator, ignorecase):
+        if ignorecase:
+            v = v.lower()
+        return [
+            int(x)
+            if x.isdigit()
+            else [int(y) if y.isdigit() else y for y in re.findall("\d+|[a-zA-Z]+", x)]
+            for x in re.split(separator, v)
+        ]
+
+    def compare(self, version2, separator=". |-", ignorecase=True):
+        # return 1 if self.version > version2
+        # return 0 if self.version == version2
+        # return -1 if self.version < version2
+        # return False if not comparable
+        if "rc" in self.version and not "rc" in version2:
+            a = self._preprocess(
+                self.version.split("rc")[0].strip("-"), separator, ignorecase
+            )
+            b = b = self._preprocess(version2, separator, ignorecase)
+            if ((a > b) - (a < b)) == 0:
+                return -1
+            else:
+                return (a > b) - (a < b)
+        else:
+            a = self._preprocess(self.version, separator, ignorecase)
+            b = self._preprocess(version2, separator, ignorecase)
+            try:
+                return (a > b) - (a < b)
+            except:
+                return False
+
+
+SUPPORTED_PLATFORM_MACHINE = {
+    "Linux": ["x86_64", "amd64", "arm64", "armv8", "aarch64"],
+    "Darwin": ["x86_64", "arm64", "arm"],
```

Comment:
`arm` is for arm 32-bit env. I don't think Darwin provides this support.

Reply:
> `arm` is for arm 32-bit env. I don't think Darwin provides this support.

https://apple.stackexchange.com/questions/420452/running-uname-m-gives-x86-64-on-m1-mac-mini 
It says that they get "arm/arm64" despite different shells and homebrew (or lack thereof) setups on each.
But we know that arm and arm64 are always 64 bit for M1 and other apple silicons. Thus I made this assumption.

Reply:
Got it. I didn't notice that. Thanks for clarification.